File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashmap = {} # num : idx
        # [1, 3, 3, 6] 
        for idx, num in enumerate(nums):
            key = target - num
            if key in hashmap:
                return  [hashmap[key], idx]
            hashmap[num] = idx
        

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> indexes = new HashMap<>();
        for (int i=0; i<nums.length; i++) {
            int val = target - nums[i];
            if (indexes.containsKey(val)) {
                return new int[]{i, indexes.get(val)};
            }
            indexes.put(nums[i], i);
        }
        return new int[] {};
    }
}